cloudinit/sources/helpers/upcloud.py

```diff
@@ -188,6 +276,13 @@ def _get_subnet_config(ip_addr, dns):
     return {"version": 1, "config": nic_configs}
 
 
+def convert_network_config(config, version=2):
+    if version == 2:
+        return convert_to_network_config_v2(config)
+
+    return convert_to_network_config_v1(config)
```

Comment:
Hi! I can get rid of version 1 entirely if all distros support it nowadays, no need to leave legacy behind (just thought some unsupported might find it useful). The problem is that I've ran into some problems with version 2 and ENI rendering in Debian (Buster) when using stateless DHCPv6 (with SLAAC).

Network config for the interface (ens5) with this config: `'ens5': {'match': {'macaddress': '3a:d6:ba:4a:29:51'}, 'dhcp6': True, 'accept_ra': True}`

Interface ends up looking like this that obviously doesn't work:
```
auto ens5
iface ens5 inet6 dhcp
```

Correct form would be this:
```
auto ens5
iface ens5 inet6 auto
    dhcp 1
```

This is rendered correctly with v1 but I didn't find separate support for SLAAC in version 2 (and thus added accept_ra). I can go digging through the conversion code, but if you know immediately where the problem lies, please let me know.

Reply:
If I understood correctly, the v2 -> v1 conversion happens with `handle_ethernets` in net/network_state.py. It's subfunction _v2_to_v1_ipcfg handles the ethernet -> subnet conversion and seems to be oblivious of both accept_ra and other DHCPv6 types. I'm gonna check that it works correctly in a pure-netplan distro, but this might be a showstopper in using v2 config for now.

Reply:
If there is a bug rendering from input in v1 or v2 to some output format (eni, sysconfig, netplan), then we need to fix that in cloud-init.   Its not really helpful to anyone to make the cloud platform have to guess based on image name or symantics or something.... If there are rendering bugs then lets fix them not push that to the cloud platform to make some symantic guess.

Reply:
Certainly so, just confirming that it wasn't just me writing erroneous Netplan config and then blaming cloud-init for it. :) I'll get rid of v1 config here and make a separate PR for the correct rendering support.

Reply:
I don't remember or know off the top of my head on the correct rendering to ENI.  I expect you're probably right and assume you came to that conclusion by actual experience.

So yeah, please go forward as you suggested.

Having a bug filed on launchpad would be good, and then the PR referencing it.

Reply:
The discovered bug was eventually an error in the datasource (accept-ra was accept_ra by mistake, causing the issue). But unfortunately the problem with ENI still remains. When an interface is configured with dhcp6 and accept-ra, ENI configuration `(inet6 auto & accept_ra 1)` causes the OS to try configuring it with stateful DHCPv6, where address is discovered through router advertisements, but it will also try to fetch the address through DHCP. End result is a working IPv6 interface, but network service fails due to this.

My solution for now is to use the network config version 1 as a default, with version 2 added for future use. Would this be acceptable or does cloud-init have a strict policy of version 2 for all new datasources? I'm all for using only that, but the lack of stateless DHCPv6 is a blocker.

Reply:
@ajmyyra V1 is perfectly acceptable in this case. Thanks for you patience in figuring all of this out.

Reply:
@ajmyyra Were you planning on updating this section? Since v2 isn't sufficient for your needs, I think you should remove the v2 conversion entirely and only support v1.

Reply:
I originally planned to leave it for future use, but might be better to remove. I'll get that done now.